python/test/unit/operators/test_matmul.py

```diff
@@ -148,9 +155,9 @@ def init_input(m, n, dtype):
             return sign | val
         if dtype == "int8":
             return torch.randint(-128, 127, (m, n), device="cuda", dtype=torch.int8)
-        dtype = {"float16": torch.float16, "bfloat16": torch.bfloat16, "float32": torch.float32}[dtype]
-        exponents = torch.randint(-10, 0, size=(m, n))
-        ret = (2.**exponents).to(dtype).to("cuda")
+        # Use small range of values to prevent numerical issues.
+        exponents = torch.randint(-4, 0, size=(m, n))
+        ret = (2.**exponents).to(getattr(torch, dtype)).to("cuda")
```

Comment:
can we use a smaller range only for the formats that need it.

Reply:
done